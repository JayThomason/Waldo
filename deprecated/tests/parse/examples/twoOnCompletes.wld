OutOfOrderOnComplete

Endpoint HighPriority;
Endpoint LowPriority;


Sequences
{
    RefreshA: HighPriority.a -> LowPriority.b -> HighPriority.b;
}

Shared
{
    Nothing controls Number sharedValue = 2;
}

MessageSequence RefreshA
{

    HighPriority.a() {}

    LowPriority.b    {}

    HighPriority.b   {}
    HighPriority.onComplete {}              
   HighPriority.onComplete {}              
}


HighPriority
{
    Public Function getValue() returns Number
    {
        return sharedValue;
    }
}

LowPriority
{
    Public Function getValue() returns Number
    {
        return sharedValue;
    }    
}

