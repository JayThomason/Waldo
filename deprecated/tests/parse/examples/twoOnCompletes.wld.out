*************************
Error: you cannot specify two onComplete functions for the same endpoint in a 
single message sequence.   
-------
Line numbers:
25
26
-------
Program text:



22        LowPriority.b    {}
23    
24        HighPriority.b   {}
25*       HighPriority.onComplete {}              
26       HighPriority.onComplete {}              
27    }
28    
29    



23    
24        HighPriority.b   {}
25        HighPriority.onComplete {}              
26*      HighPriority.onComplete {}              
27    }
28    
29    
30    HighPriority
*************************



